src/client/Microsoft.Identity.Client/Kerberos/KerberosSupplementalTicketManager.cs

```diff
@@ -0,0 +1,240 @@
+            }
+        }
+
+        /// <summary>
+        /// Reads a Kerberos Service Ticket associated with given service principal name from
+        /// current user's Ticket Cache.
+        /// </summary>
+        /// <param name="servicePrincipalName">Service principal name to find associated Kerberos Ticket.</param>
+        /// <returns>Byte stream of searched Kerberos Ticket information if exists. Null, otherwise.</returns>
+        /// <remarks>
+        /// Can throws <see cref="Win32Exception"/> if error occurs while searching ticket information from Ticket Cache.
+        /// </remarks>
+        public static byte[] GetKerberosTicketFromWindowsTicketCache(string servicePrincipalName)
+        {
+            return GetKerberosTicketFromWindowsTicketCache(servicePrincipalName, _defaultLogonId);
+        }
+
+        /// <summary>
+        /// Reads a Kerberos Service Ticket associated with given service principal name from
+        /// current user's Ticket Cache.
+        /// </summary>
+        /// <param name="servicePrincipalName">Service principal name to find associated Kerberos Ticket.</param>
+        /// <param name="logonId">The Logon Id of the user owning the ticket cache.
+        /// The default of 0 represents the currently logged on user.</param>
+        /// <returns>Byte stream of searched Kerberos Ticket information if exists. Null, otherwise.</returns>
+        /// <remarks>
+        /// Can throws <see cref="Win32Exception"/> if error occurs while searching ticket information from Ticket Cache.
+        /// </remarks>
+        public static byte[] GetKerberosTicketFromWindowsTicketCache(string servicePrincipalName, long logonId)
+        {
+            if (!IsWindows())
+            {
+                throw new NotSupportedException("Ticket Cache interface is not supported for this OS platform.");
+            }
+
+            using (var reader = new Win32.TicketCacheReader(servicePrincipalName, logonId))
+            {
+                return reader.RequestToken();
+            }
+        }
+
+        /// <summary>
+        /// Gets the KRB-CRED Kerberos Ticket information as byte stream.
+        /// </summary>
+        /// <param name="ticket">Kerberos ticket object to save.</param>
+        /// <returns>Byte stream representaion of KRB-CRED Kerberos Ticket if it contains valid ticket information.
+        /// Null, otherwise.</returns>
+        public static byte[] GetKrbCred(KerberosSupplementalTicket ticket)
+        {
+            if (!string.IsNullOrEmpty(ticket.KerberosMessageBuffer))
+            {
+                return Convert.FromBase64String(ticket.KerberosMessageBuffer);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        ///  Checks the current application is running under Windows OS or not.
+        /// </summary>
+        /// <returns>True if running under Windows OS. False, otherwise.</returns>
+        internal static bool IsWindows()

```

Comment:
I just realized that we already have this functionality here
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/0d53f3e80e29f9c6d172e069c10e02d71e24f65e/src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/DesktopOsHelper.cs#L23

Reply:
I'd tried DesktopOsHelper.IsWindows() at the first time, but it is not working for net45 environment and compiler complains there's missed assembly reference.  Checking all used cases, it's only working under NET_CORE or DESKTOP condition. With this reason, I'd implemented this function with reference for it.